examples/first.cpp

```diff
@@ -1,59 +1,186 @@
 #include "loops/loops.hpp"
+#include <algorithm>
 #include <vector>
 #include <iostream>
 
-typedef int (*justcompiled_t)(const int* ptr, int64_t n,
+typedef int (*minmaxpos_t)(const int* ptr, int64_t n,
                             int* minpos, int* maxpos);
-loops::Func gencode(loops::Context& ctx)
+loops::Func genminmaxloc(loops::Context& CTX)
 {
     using namespace loops;
     IReg ptr, n, minpos_addr, maxpos_addr;
-    ctx.startFunc("foo", {&ptr, &n, &minpos_addr, &maxpos_addr});
-    IReg i = ctx.const_(0);
-    IReg minpos = ctx.const_(0);
-    IReg maxpos = ctx.const_(0);
+    CTX.startFunc("minmaxloc", {&ptr, &n, &minpos_addr, &maxpos_addr});
+    IReg i = CTX.const_(0);
+    IReg minpos = CTX.const_(0);
+    IReg maxpos = CTX.const_(0);
     IReg minval = load_<int>(ptr);
     IReg maxval = minval;
-    IReg elemsize = ctx.const_((int)sizeof(int));
-    n *= elemsize;
-    ctx.do_();
+    n *= sizeof(int);
+    CTX.do_();
```

Comment:
since it will be a long time (probably, 2-3 weeks at least) while the next PR is submitted with Neon support, I suggest to modify this PR or submit a new one (before Neon) with the CTX-less syntax for control flow

Reply:
Done.